src/App.tsx

```diff
@@ -17,18 +77,19 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+            <TodoFilter onSearchChange={handleSearchChange} onStatusChange={handleStatusChange}/>
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+            {loading ? <Loader /> : <TodoList todos={filteredTodos} openModal={openModal} selectedTodoId={selectedTodoId} />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {showModal && selectedTodoId && (
```

Comment:
```suggestion
      {selectedTodoId && (
```
showModal state also looks redundant here